modules/dns_dnscmd/dns_dnscmd_main.rb

```diff
@@ -33,19 +31,17 @@ def do_remove(name, type)
       nil
     end
 
-    def execute(cmd, msg = nil, error_only = false)
+    def execute(*args, **keyword_args)
       tsecs = 5
       response = nil
-      interpreter = Proxy::SETTINGS.x86_64 ? 'c:\windows\sysnative\cmd.exe' : 'c:\windows\system32\cmd.exe'
-      command = interpreter + ' /c c:\Windows\System32\dnscmd.exe ' + "#{@server} #{cmd}"
-
       std_in = std_out = std_err = nil
+      real_cmd = ['c:\Windows\System32\dnscmd.exe' + "#{@server}"] + args
       begin
         timeout(tsecs) do
-          logger.debug "executing: #{command}"
-          std_in, std_out, std_err = Open3.popen3(command)
-          response = std_out.readlines
-          response += std_err.readlines
+          logger.debug { "executing: #{real_cmd}" }
+          std_in, std_out, std_err = popen3(real_cmd)
```

Comment:
Also, you cannot pass array into popen3 method, it will pass it as an array, that will not work:

```
>> x, y, z = Open3.popen3(["/usr/bin/bash", "-c", "echo hello"])
Traceback (most recent call last):
        8: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `<main>'
        7: from /home/lzap/.rbenv/versions/2.7.2/bin/irb:23:in `load'
        6: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        5: from (irb):9
        4: from (irb):10:in `rescue in irb_binding'
        3: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/open3.rb:101:in `popen3'
        2: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/open3.rb:213:in `popen_run'
        1: from /home/lzap/.rbenv/versions/2.7.2/lib/ruby/2.7.0/open3.rb:213:in `spawn'
ArgumentError (wrong first argument)
```

You need to unwrap the array here:

```
>> x, y, z = Open3.popen3("/usr/bin/bash", "-c", "echo hello")
=> [#<IO:fd 8>, #<IO:fd 10>, #<IO:fd 12>, #<Process::Waiter:0x0000000000e53ee0 run>]
```

This can be done with:

```suggestion
          std_in, std_out, std_err = popen3(*real_cmd)
```